python/simulators/CMakeLists.txt

```diff
@@ -18,6 +18,33 @@ set(PYTHON_OPM_SIMULATORS_PACKAGE_PATH ${PROJECT_BINARY_DIR}/python/opm/simulato
 set_target_properties( simulators PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${PYTHON_OPM_SIMULATORS_PACKAGE_PATH} )
 
 target_link_libraries( simulators PRIVATE opmsimulators )
+# Add the binary (build) directory to the include directories for the target
+
+# Set the path to the input docstrings.json file and the output .hpp file
+set(PYTHON_DOCSTRINGS_FILE "${PROJECT_SOURCE_DIR}/python/docstrings.json")
+set(PYTHON_DOCSTRINGS_GENERATED_HPP "${PROJECT_BINARY_DIR}/python/PyBlackOilSimulatorDoc.hpp")
+# Command to run the Python script
+add_custom_command(
+    OUTPUT ${PYTHON_DOCSTRINGS_GENERATED_HPP}
+    COMMAND ${CMAKE_COMMAND} -E env PYTHONPATH=${CMAKE_SOURCE_DIR}
+            ${PYTHON_EXECUTABLE} ${PROJECT_SOURCE_DIR}/python/generate_docstring_hpp.py
+            ${PYTHON_DOCSTRINGS_FILE} ${PYTHON_DOCSTRINGS_GENERATED_HPP}
+    DEPENDS ${PYTHON_DOCSTRINGS_FILE}
+    COMMENT "Generating PyBlackOilSimulatorDoc.hpp from JSON file"
+)
+
+# Create a custom target that depends on the generated .hpp file
+add_custom_target(
```

Comment:
this target is unnecessary. just add the generated hpp to the sources of simulators instead.